tests/test_api/test_security_hardening.py

```
@@ -0,0 +1,219 @@
+
+    def test_creates_directory_with_0700(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            path = f"{tmp}/test_cache"
+            ensure_secure_directory(path)
+            mode = stat.S_IMODE(Path(path).stat().st_mode)
+            assert mode == 0o700, f"Expected 0700, got {oct(mode)}"
+
+    def test_tightens_existing_loose_permissions(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            path = f"{tmp}/test_cache"
+            os.makedirs(path, mode=0o755)
```

Ruff (0.16.3)

[warning] `os.makedirs()` should be replaced by `Path.mkdir(parents=True)`

(PTH103)